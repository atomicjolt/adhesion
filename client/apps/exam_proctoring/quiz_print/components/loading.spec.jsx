import React            from 'react';
import TestUtils        from 'react-addons-test-utils';
import Loading          from './loading';
import Stub             from '../../../../specs_support/stub';

describe('loading', () => {
  let result;
  let props;

  beforeEach(() => {
    props = {
      loadingQuiz: true,
    };
    result = TestUtils.renderIntoDocument(<Stub><Loading {...props} /></Stub>);
  });

  it('Says its loading', () => {
    const element = TestUtils.scryRenderedDOMComponentsWithTag(result, 'div')[0];
    expect(element.textContent).toContain('Loading Quiz');
  });

  it('loading questions', () => {
    props.loadingQuiz = false;
    result = TestUtils.renderIntoDocument(<Stub><Loading {...props} /></Stub>);
    const element = TestUtils.scryRenderedDOMComponentsWithTag(result, 'div')[0];
    expect(element.textContent).toContain('Loading Questions');
  });
});
