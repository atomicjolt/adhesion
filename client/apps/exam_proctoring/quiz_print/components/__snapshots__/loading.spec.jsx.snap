// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loading matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <loading
    loadingQuiz={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <_DefaultExportValue />,
        <h3>
          Loading 
          Quiz
        </h3>,
      ],
      "style": Object {
        "textAlign": "center",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Loading ",
            "Quiz",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Loading ",
          "Quiz",
        ],
        "type": "h3",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <_DefaultExportValue />,
          <h3>
            Loading 
            Quiz
          </h3>,
        ],
        "style": Object {
          "textAlign": "center",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Loading ",
              "Quiz",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Loading ",
            "Quiz",
          ],
          "type": "h3",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
